File: CS/MyButtonEdit/RepositoryItemMyButtonEdit.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Drawing;

namespace WindowsApplication1
{
    [UserRepositoryItem("Register")]
    public class RepositoryItemMyButtonEdit : RepositoryItemButtonEdit
    {

        static RepositoryItemMyButtonEdit()
        {
            Register();
        }
        public RepositoryItemMyButtonEdit() 
        {
            TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor;
        }

        public override void CreateDefaultButton()
        {
        }



        internal const string EditorName = "MYButtonEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(MyButtonEdit),
                typeof(RepositoryItemMyButtonEdit), typeof(DevExpress.XtraEditors.ViewInfo.ButtonEditViewInfo),
                new MyButtonEditPainter(), true, null));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }

        public override void Assign(RepositoryItem item)
        {
            base.Assign(item);
            TextToScroll = ((RepositoryItemMyButtonEdit)item).TextToScroll;
        }

        private string _TextToScroll;

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
        public string TextToScroll
        {
            get { return _TextToScroll; }
            set { _TextToScroll = value; }
        }

        public override string GetDisplayText(object editValue)
        {
            return  TextToScroll;
        }

        public override string GetDisplayText(DevExpress.Utils.FormatInfo format, object editValue)
        {
            return TextToScroll;
        }

    }
}

File: VB/MyButtonEdit/RepositoryItemMyButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Drawing

Namespace WindowsApplication1
	<UserRepositoryItem("Register")> _
	Public Class RepositoryItemMyButtonEdit
		Inherits RepositoryItemButtonEdit

		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
			TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor
		End Sub

		Public Overrides Sub CreateDefaultButton()
		End Sub



		Friend Const EditorName As String = "MYButtonEdit"

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(MyButtonEdit), GetType(RepositoryItemMyButtonEdit), GetType(DevExpress.XtraEditors.ViewInfo.ButtonEditViewInfo), New MyButtonEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property

		Public Overrides Sub Assign(ByVal item As RepositoryItem)
			MyBase.Assign(item)
			TextToScroll = (CType(item, RepositoryItemMyButtonEdit)).TextToScroll
		End Sub

		Private _TextToScroll As String

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
		Public Property TextToScroll() As String
			Get
				Return _TextToScroll
			End Get
			Set(ByVal value As String)
				_TextToScroll = value
			End Set
		End Property

		Public Overrides Overloads Function GetDisplayText(ByVal editValue As Object) As String
			Return TextToScroll
		End Function

		Public Overrides Overloads Function GetDisplayText(ByVal format As DevExpress.Utils.FormatInfo, ByVal editValue As Object) As String
			Return TextToScroll
		End Function

	End Class
End Namespace